nautobot/extras/datasources/git.py

```diff
@@ -288,6 +291,14 @@ def ensure_git_repository(
     from_url, to_path, from_branch = get_repo_from_url_to_path_and_from_branch(
         repository_record, logger=logger, job_result=job_result
     )
+    # We want to check if the repo is already checked out at head. We also want to avoid calling
+    # nautobot.utilities.git.GitRepo.checkout, because it will cause the URL to be rebuilt causing calls to a secrets
+    # backend. As such, if head is None, we can't perform these checks.
+    if head is not None:
+        # If the repo exists and has HEAD already checked out, the repo is present and has the correct branch selected.
+        with suppress(InvalidGitRepositoryError):
+            if Path(to_path).exists() and Repo(to_path).head == head:
+                return
```

Comment:
To prevent unnecessary calls to `repository_record.secrets_group.get_secret_value()` this should be moved to the top of `ensure_git_repository` and use `repository_record.filesystem_path` instead of `to_path`

Reply:
Good call, thank!